jujugui/static/gui/src/app/store/env/controller-api.js

```diff
@@ -1261,6 +1261,96 @@ YUI.add('juju-controller-api', function(Y) {
         request: 'RevokeCredentials',
         params: {entities: [{tag: tags.build(tags.CREDENTIAL, name)}]}
       }, handler);
+    },
+
+    /**
+      Modify (grant or revoke) user access to the specified model.
+
+      @method grantUserAccess
+      @param {Array} userIds The IDs of users who need modified access.
+      @param {String} modelId The ID for the model on which the access is being
+                              changed.
+      @param {String} access The level of access to grant the users; can be
+                             'read', 'write', or 'admin'.
+      @param {String} action Either 'revoke' or 'grant'.
+      @param {Function} callback A callable that must be called once the
+        operation is performed. It will receive an error message or null if the
+        access modification succeeded.
+    */
+    _modifyModelAccess: function(users, modelId, access, action, callback) {
+      // Decorate the user supplied callback.
+      const handler = data => {
+        if (!callback) {
+          console.log(
+            'Data returned by ModelManager.ModifyModelAccess API call:', data);
+          return;
+        }
+        if (data.error) {
+          callback(data.error);
+          return;
+        }
+        const results = data.response.results;
+        if (!results || results.length !== 1) {
+          // This should never happen.
+          callback('Invalid results from Juju: ' + JSON.stringify(results));
```

Comment:
s/Invalid/invalid/ to mimic Juju error style, which allows for concatenating error messages.